pkg/client/outofband/client.go

```diff
@@ -63,17 +70,26 @@ type Event interface {
 type MessageOption func(*message) error
 
 type message struct {
-	Label    string
-	Goal     string
-	GoalCode string
-	Service  []interface{}
+	Label             string
+	Goal              string
+	GoalCode          string
+	RouterConnections []string
+	Service           []interface{}
+}
+
+func (m *message) RouterConnection() string {
+	if len(m.RouterConnections) == 0 {
+		return ""
+	}
+
+	return m.RouterConnections[0]
```

Comment:
Unlike didexchange invitations, an OOB invitation's `service` attribute can contain multiple entries, so it makes sense to return all router connection IDs and use them

Reply:
It leads us to `chooseTarget` function. Which picks up the first one.

Reply:
There is a separation between what the client includes in an OOB invitation, and what the other agent (that may not even be based on this framework) does with it.

`chooseTarget` is what this framework is doing. Another framework may not have the same logic.

Reply:
We should also provide the user with the ability to choose with `service` entry they would like to use when accepting an OOB invitation.

Reply:
Opened #2209

Reply:
This issue can be deferred for a followup

Reply:
Ok